tests/cases/auth_context.rs

```diff
@@ -0,0 +1,293 @@
+
+static CLIENT_CRT: &str = r#"-----BEGIN CERTIFICATE-----
+MIIDRzCCAi8CFBH6c/YUjvTe8EmdS5jhnePyW3XxMA0GCSqGSIb3DQEBCwUAMGAx
+CzAJBgNVBAYTAkZSMQ8wDQYDVQQIDAZGcmFuY2UxDjAMBgNVBAcMBVBhcmlzMRAw
+DgYDVQQKDAdncnBjLXJzMR4wHAYDVQQDDBVncnBjLXJzIHRlc3RzIFJvb3QgQ0Ew
+IBcNMTkwNTA3MDkzOTE5WhgPMzAxODA5MDcwOTM5MTlaMF4xCzAJBgNVBAYTAkZS
+MQ8wDQYDVQQIDAZGcmFuY2UxDjAMBgNVBAcMBVBhcmlzMRAwDgYDVQQKDAdncnBj
+LXJzMRwwGgYDVQQDDBNncnBjLXJzIFRlc3QgQ2xpZW50MIIBIjANBgkqhkiG9w0B
+AQEFAAOCAQ8AMIIBCgKCAQEA0zxpjVNg4kntLL31VR/zQO4YE6MtyqJOBVKbeCc6
+R15KLYC5zjz4NchPouquJX23JmLoLtMgBnoIlEC6LbwckqxrYoOrT8FfBWyIHMrI
+LGGQtQv/1wFKtOsbZugaRoUpklIkv6h2kOHO9Sj7TKpVRzNLXADUtiCClzxncg1T
+z98G/PrbCw7LZB09S9KiJnUV6QEHuhqiG0ROrSW+/rH5IHjtsaFDh9IBQNCkneMD
+D2KfbpQ/k/Z+8KckBSNpPRRLIioG6ZJGu0W/ZNHW7/CvHfxnM1lW+w4aNcmW0FZv
+zAP1sA2GHfYp7N3QXt0ORdPjyOtQwM0xS3aPf1CaCEj5DwIDAQABMA0GCSqGSIb3
+DQEBCwUAA4IBAQAQGVYG3wXu1EZuQTwW8ec0ROwKEoJltKsoRAvHHXvwpEOCmDyl
+8Gxk5j+O8Xytc4lBcViAz8EJ6senGk1RdNiSQLAf0dGZPsfuCGrp13gGG4ZbMXZX
+oI4Z2lTvxTEVMMpAoHHQheYRYZk1z3gKHeKuiu4tAiti39iCS1SninrINBlvU5ZS
+ijUszQEhmGoYb2OxjBHw8iYwE9GC33rCNWsONGXGkM7wNt3A64SYwOsXXT3wG8CX
+kqX0HFP2JaX7x0BqG1Jh0VZ+QoEKmfPOREv20q8aO6+s0mEu+88eGpnJWlBXhtfD
+LOqbHJD2Bktma6qlvdjMFHgkJVon+Z1pIoes
+-----END CERTIFICATE-----"#;
+
+static CLIENT_KEY: &str = r#"-----BEGIN RSA PRIVATE KEY-----
+MIIEogIBAAKCAQEA0zxpjVNg4kntLL31VR/zQO4YE6MtyqJOBVKbeCc6R15KLYC5
+zjz4NchPouquJX23JmLoLtMgBnoIlEC6LbwckqxrYoOrT8FfBWyIHMrILGGQtQv/
+1wFKtOsbZugaRoUpklIkv6h2kOHO9Sj7TKpVRzNLXADUtiCClzxncg1Tz98G/Prb
+Cw7LZB09S9KiJnUV6QEHuhqiG0ROrSW+/rH5IHjtsaFDh9IBQNCkneMDD2KfbpQ/
+k/Z+8KckBSNpPRRLIioG6ZJGu0W/ZNHW7/CvHfxnM1lW+w4aNcmW0FZvzAP1sA2G
+HfYp7N3QXt0ORdPjyOtQwM0xS3aPf1CaCEj5DwIDAQABAoIBAGqkLy2YEGttov1Z
+kUe+b5IyIZmYYf/Q3M63G6IXO8bYBV7fg+5ovW55iCrZi72FsDcC6/DnyV07sqUV
+4GhCdbJetX/wtUX4e75DQpw2i8RMJUAhpLGpB0w6/YSAXPOJOqmSKtdPdzxBo7gF
+Tokv3QmrDbqO+NRRqVwstJTRwaRJK3KpVIuCbB2JRoGfY9ztDnWFd3qmbFTpeVzA
+Bd3zvRuHtMWmxYtY6mQ9ZwCO+3zw2/6Vj6eBjNd/EMm4plbZXMye7ih7IzFnswub
+q+6jkRJNnICF17En4CYBLyMNmpeFgczqZiRhMv/2s/Z1t6d0SCE29QB8bpApbnEb
+7f56e/ECgYEA6YIm+3OkrZCodM0LoDG3sDRMEaImZWWqkyDeS6I3oG69a+QgZNlJ
+thA89EqqTTYryV5Y7Px7dDNVVS+uE39S+5y02Vu8uC1lq3eqJe9aempOQo7zI7XO
++LEGHuH8ufN+pOARDX+46G99q0liXheIuuYcI3MPFE8ZAFdVtgxasIUCgYEA55UR
+IP8+O5mR0EmfRbPM5HCyE0NA3GI9ONSVDVX/PG1rl7xMhlXH3cYYo4sOB8nMcIWc
+mHPJT5gqTGvgwrv6UpXg7FEV42ILL+z6sbY0AT/EvuDzFNbaQYV++8pJ0gg+jfc2
+nguyoxyFXMTPvKjykc8l2YrzLwpS5HL/xaZ5oYMCgYBaSj5CioPJrR7GEycEqVTK
+qF4lj1pjaL1dKJ9sBWyW91r9Jfe0pyROZtb66wgQ1Pp3Yajm8fXWh6beSpJ37Yqr
+yftP9J3zZXJjq+C5zGRi40ohiF7RlxJRrTmdpWpwiIuWuyE84/8euUWRfs2vblAr
+XKDSwEG7a55SOHLm8r36dQKBgA/S0a7NgfUUkzAGN8nNdUyOnrxONhpY5BIEBIM6
+wwvf9iASD7CGH/f5VymlkLnbggqWNpL8Jl836Hv8SwlDSuFluE6tY697vYPyz1Zh
+OsBzGQBp4nIkFBmyA3E060JcqdmNcL/v0K7wffROxlr/uENaQh3RLriiHfPaH7vj
+mh/9AoGAEqDij1mu1KtwgjDo/2vRZDC8/iWARPdkM+/5KX+NQyEueQ7DVCJ395of
+yq+fZMhZpCMrnHsTIbTa3T1+GMzghqYASI0DJPLk9z18AeMP/5rWCe8KjitTvYDH
+iEmsL1oMGCGH/W5H68gboDXxWcvS8PE7gA4NrNJhLzLorRF8SoQ=
+-----END RSA PRIVATE KEY-----"#;
+
+// CA key:
```

Comment:
Why comment them out?

Reply:
The CA root private key is not used by any piece of code (and will probably never be, because I doubt the test is ever going to generate its own set of keys). The compiler does not need to bother about it.
If however someone ever needs to generate another different key when adding functionality, they may need that key (or would need to regenerate everything), which is why I made it available.